src/legacy/core_plugins/metrics/public/components/vis_editor.js

```diff
@@ -66,7 +69,39 @@ export class VisEditor extends Component {
     this.props.vis.updateState();
   }, VIS_STATE_DEBOUNCE_DELAY);
 
-  handleChange = async (partialModel) => {
+  isValidKueryQuery = (filterQuery) => {
```

Comment:
I think we need to re-think how to handle invalid KQL queries. There are a few issues I'm seeing with the current approach:

* The vis editor now needs special knowledge of the shape of the model objects of all of its subcomponents. This will make the validation likely to break if the shape of those model objects ever change as it won't be obvious that this file is relying on the previous shape.
* A KQL validation error now silently prevents any updates in the editor from happening even if those updates are unrelated to the query. I think this will confuse users.
* I'm still getting a 500 error if I put an invalid KQL query in the "Data" tab and then switch to the "Panel Options" tab and make a change.

I propose we handle KQL errors in the same way Lucene errors are handled. If you enter an invalid lucene query it still gets sent to the server. The server responds with a 200 no matter what, but it reports the syntax error in the response payload. Then the frontend can decide whether to report the error to the user or silently ignore it (we should report it). This screenshot should show what I'm talking about:

![image](https://user-images.githubusercontent.com/6239176/58838202-a8792600-8623-11e9-8a81-92ab8623f4db.png)

Reply:
I agree with the first comment on 

> vis editor now needs special knowledge of the shape of the model objects of all of its subcomponents
However, I feel this should be handled if and when the situation arises. We do not know if TSVB will continue to be supported or if it will be sun-setted. If the shape of the model and its subcomponents change, then a lot more work will be needed throughout TSVB to accommodate that and there will be more validation methods (on the client and on the server side for request preprocessing) that will fail. 

I also agree with your second comment on handling how invalid-syntax KQL queries are not handled in the same manner as lucene queries. I suggest that the work should be a separate issue and PR. We can make the changes to TSVB once the work is done.

Reply:
The updates are handles in a commit.